src/main/java/seedu/address/model/util/DateTime.java

```diff
@@ -0,0 +1,83 @@
+package seedu.address.model.util;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAccessor;
+
+public class DateTime implements Comparable<DateTime> {
+    public static final String TIME_REGEX = "(([0-1]\\d)|(2[0-3])):([0-5]\\d)";
+    public static final String DATE_REGEX = "(([0-2]\\d)|(3[0-1]))-((0[1-9])|(1[0-2]))-(\\d{4})";
+    public static final String MESSAGE_CONSTRAINTS =
+            "From should be in the format of DD-MM-YYYY or DD-MM-YYYY HH:mm, and should not be blank."
```

Comment:
From, To and Deadline will implement this so Message might need to be more general